pkg/config/derived-service-cert.go

```
@@ -92,10 +114,33 @@ func (idCfg *IdentityConfig) derivedServiceCertConfig() error {
 	}
 
 	// k8s-athenz-sia uses third-party service certificate, instead of using CopperArgos:
-	if idCfg.KeyFile != "" && idCfg.CertFile != "" {
-		idCfg.ServiceCert.LocalCert = LocalCertMode{Use: true}
+	if len(keyPaths) > 0 && len(certPaths) > 0 {
+		if len(certPaths) > 1 || len(keyPaths) > 1 {
+			log.Warn("Multiple cert/key paths provided. Using the first path.")
+		}
+		idCfg.ServiceCert.LocalCert = LocalCertMode{
+			Use:      true,
+			CertPath: certPaths[0],
+			KeyPath:  keyPaths[0],
+		}
 		return nil // Use LocalCert Mode
 	}
 
 	return nil
 }
+
+// splitAndFilterPaths splits a comma-separated string, trims spaces, and filters out empty elements.
+func splitAndFilterPaths(s string) []string {
+	if s == "" {
+		return nil
```

Can we throw an error instead if "" found?

---

fixed in https://github.com/AthenZ/k8s-athenz-sia/pull/192/commits/3aa34307ea4042f642a86ff2c43f9a11620a0398